File: app/src/main/java/com/qiscus/rtc/sample/simple/IncomingCallActivity.java
package com.qiscus.rtc.sample.simple;

import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import android.util.Log;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.qiscus.meet.QiscusMeet;
import com.qiscus.rtc.sample.R;
import com.qiscus.sdk.Qiscus;
import com.qiscus.sdk.chat.core.data.remote.QiscusPusherApi;

import org.json.JSONObject;

public class IncomingCallActivity extends AppCompatActivity {

    private ImageView ivCallerAvatar;
    private TextView tvCallerName;
    private Button btnAcceptCall, btnDeniedCall;

    private String roomId, callerAvatar, callerDisplayName;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_incoming_call);

        roomId = getIntent().getStringExtra("roomId");
        callerAvatar = getIntent().getStringExtra("callerAvatar");
        callerDisplayName = getIntent().getStringExtra("callerDisplayName");

        ivCallerAvatar = findViewById(R.id.iv_caller_avatar);
        tvCallerName = findViewById(R.id.tv_caller_name);
        btnAcceptCall = findViewById(R.id.btn_accept_call);
        btnDeniedCall = findViewById(R.id.btn_denied_call);

        tvCallerName.setText(callerDisplayName);
        Glide.with(this).load(callerAvatar).into(ivCallerAvatar);

        btnAcceptCall.setOnClickListener(v -> {
            onAccept();
        });

        btnDeniedCall.setOnClickListener(v -> {
            onDenied();
        });
    }

    private void onAccept() {
        QiscusMeet.call()
                .setRoomId(roomId)
                .setDisplayName(Qiscus.getQiscusAccount().getUsername())
                .build(this);

        finish();
    }

    private void onDenied() {
        JSONObject json = new JSONObject();
        try {
            json.put("sender", roomId);
            json.put("event", "rejected");
            json.put("active", false);

            QiscusPusherApi.getInstance().setEvent(Long.parseLong(roomId), json);

        } catch (Exception ex) {
            Log.e("IncomingCallActivity", ex.getMessage());
        }

        this.finish();

    }
}

File: meet/src/main/java/com/qiscus/meet/IncomingCallActivity.kt
package com.qiscus.meet

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log
import com.bumptech.glide.Glide
import com.qiscus.meet.model.Call
import kotlinx.android.synthetic.main.activity_incoming_call.*
import org.jitsi.meet.sdk.JitsiMeetActivity
import org.jitsi.meet.sdk.JitsiMeetConferenceOptions

class IncomingCallActivity : AppCompatActivity() {

    private lateinit var call: Call

    companion object {
        fun generateIntent(ctx: Context, call: Call): Intent {
            val intent = Intent(ctx, IncomingCallActivity::class.java)
            intent.putExtra("call", call)
            return intent
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_incoming_call)

        call = intent.getParcelableExtra("call")
        tv_caller_name.text = call.callerDisplayName
        Log.e("debug", call.callerAvatar)
        Glide.with(this).load(call.callerAvatar).into(iv_caller_avatar)
        btn_denied_call.setOnClickListener { onDenied() }
        btn_accept_call.setOnClickListener { onAccept() }
    }

    private fun onAccept() {
        val options = JitsiMeetConferenceOptions.Builder()
            .setRoom(call.roomId)
            .build()

        JitsiMeetActivity.launch(this, options)
        finish()
    }

    private fun onDenied() {
        this.finish()
    }
}
